wp-content/plugins/core/src/Theme/Config/Web_Fonts.php

```diff
@@ -5,130 +5,101 @@
 
 
 class Web_Fonts {
-	public const TYPEKIT_ID = 'typekit';
-	public const GOOGLE     = 'google';
-	public const CUSTOM     = 'custom';
-
-
-	/**
-	 * @var string
-	 */
-	private $plugin_file;
+	public const  PROVIDER_TYPEKIT = 'typekit';
+	public const  PROVIDER_GOOGLE  = 'google';
+	public const  PROVIDER_CUSTOM  = 'custom';
+	private const TYPEKIT_API      = 'https://use.typekit.net';
+	private const GOOGLE_API       = 'https://fonts.googleapis.com';
 
 	/**
 	 * @var array
 	 */
 	private $fonts;
 
-	public function __construct( string $plugin_file, array $fonts = [] ) {
-		$this->plugin_file = $plugin_file;
+	public function __construct( array $fonts = [] ) {
 		$this->fonts       = $fonts;
 	}
 
 	/**
-	 * Add Typekit to Editor
+	 * Enqueue any required web fonts
 	 *
-	 * @filter mce_external_plugins
+	 * @action wp_enqueue_scripts
+	 * @action login_enqueue_scripts
 	 */
-	public function add_typekit_to_editor( $plugins ) {
-		if ( ! empty( $this->fonts[ self::TYPEKIT_ID ] ) ) {
-			$plugins[ self::TYPEKIT_ID ] = plugins_url( 'assets/admin/editor/typekit.tinymce.js', $this->plugin_file );
+	public function enqueue_fonts(): void {
+		foreach ( $this->get_font_urls() as $provider => $url ) {
+			wp_enqueue_style( $provider, $url, [], null, 'all' );
 		}
-
-		return $plugins;
 	}
 
 	/**
-	 * Localize Typekit TinyMCE
+	 * Unsupported Browser Page Fonts.
 	 *
-	 * @filter admin_head
+	 * @action tribe/unsupported_browser/head
 	 */
-	public function localize_typekit_tinymce() {
-		if ( empty( $this->fonts[ self::TYPEKIT_ID ] ) ) {
-			return;
+	public function inject_unsupported_browser_fonts(): void {
+		foreach ( $this->get_font_urls() as $url ) {
+			printf( "<link rel='stylesheet' href='%s' type='text/css' media='all'>\n\t", esc_url( $url ) );
 		}
-
-		printf( '<script type="text/javascript">var tinymce_typekit_id = \'%s\';</script>', $this->fonts[ self::TYPEKIT_ID ] );
 	}
 
 	/**
-	 * Add any required fonts
+	 * Visual Editor Fonts
 	 *
-	 * @action wp_head
-	 * @action login_head
+	 * @action after_setup_theme
 	 */
-	public function load_fonts() {
+	public function add_visual_editor_fonts(): void {
+		foreach( $this->get_font_urls() as $url ) {
+			add_editor_style( $url );
```

Comment:
This is only going to load styles for the MCE editor, not the block editor. Do we need these fonts in the block editor, too?

Reply:
Good catch. I though the add_editor_style was working for both. It was not. I've updated to accommodate.